src/managers/CursorManager.cpp

```diff
@@ -133,30 +136,25 @@ void CCursorManager::setCursorSurface(SP<CWLSurface> surf, const Vector2D& hotsp
 void CCursorManager::setXCursor(const std::string& name) {
     float scale = std::ceil(m_fCursorScale);
 
-    if (!xcursor.themeLoaded) {
-        Debug::log(ERR, "XCursor failed to find theme in setXCursor");
-        if (!xcursor.defaultCursor) {
-            g_pPointerManager->resetCursorImage();
-            return;
-        }
-    }
-
-    auto& icon = xcursor.defaultCursor;
-    // try to get an icon we know if we have one
-    for (auto const& c : xcursor.cursors) {
-        if (c.first != name)
-            continue;
+    m_currentXcursor = m_pXcursor->getShape(name);
+    auto icon        = m_currentXcursor->images.front();
 
-        icon = c.second;
-    }
+    m_vCursorBuffers.emplace_back(makeShared<CCursorBuffer>((uint8_t*)icon.pixels.data(), icon.size, icon.hotspot));
 
-    m_vCursorBuffers.emplace_back(makeShared<CCursorBuffer>((uint8_t*)icon->pixels.data(), icon->size, icon->hotspot));
-
-    g_pPointerManager->setCursorBuffer(getCursorBuffer(), icon->hotspot / scale, scale);
+    g_pPointerManager->setCursorBuffer(getCursorBuffer(), icon.hotspot / scale, scale);
     if (m_vCursorBuffers.size() > 1)
         dropBufferRef(m_vCursorBuffers.at(0).get());
 
     m_bOurBufferConnected = true;
+
+    if (m_currentXcursor->images.size() > 1) {
+        // animated
+        wl_event_source_timer_update(m_pAnimationTimer, m_currentXcursor->images[0].delay);
```

Comment:
please use our timers (see eventloopmgr)

Reply:
was basing it of the hyprcursor animation ticker but okay il try

Reply:
8de3d591ec4985c9cc004562ed4a9bf0a2f7e23b should do it